dlib/vectorstream/vectorstream.h

```diff
@@ -120,15 +146,12 @@ namespace dlib
             std::iostream(&buf),
             buf(buffer)
         {}
-
-        std::istream& seekg (
-            std::streampos pos
-        )
-        {
-            buf.seekg(pos);
-            return *this;
-        }
-
+            
+        vectorstream(const vectorstream& ori) = delete;
+        vectorstream(vectorstream&& item) = delete;
+        vectorstream& operator=(const vectorstream& ori) = delete;
+        vectorstream& operator=(vectorstream&& item) = delete;
```

Comment:
You don't need to delete `&&` versions.  They are deleted by default when the normal ones are deleted.

Reply:
So shall we just make ```vectorstream``` inherit from```dlib::noncopyable``` ?

Reply:
Leave the two deleted ones. They will give slightly better compile time errors than what users receive if noncopyable is used.